var/spack/repos/builtin/packages/neurodamus-model/package.py

```diff
@@ -131,13 +153,24 @@ def install(self, spec, prefix):
             sed('-i', 's#-dll .*#-dll %s#' % prefix.lib.join('libnrnmech.so'),
                 prefix.bin.special)
 
-        # PY: Link only important stuff, and create a new lib link (to our lib)
-        py_src = spec['neurodamus-core'].prefix.python
-        if os.path.isdir(py_src):
-            os.makedirs(prefix.python)
-            force_symlink('../lib', prefix.python.lib)
+        if spec.satisfies('+coreneuron'):
+            # move special-core to bin and libcorenrnmech* to lib
+            outdir = spec.architecture.target
+            shutil.move(join_path(outdir, 'special-core'), prefix.bin)
+            for libname in find_libraries("libcorenrnmech*", outdir):
+                shutil.move(libname, prefix.lib)
+            # Then modc
```

Comment:
@ferdonline  : could you post the output of `ldd special-core` after install? I wonder if paths are getting resolved if we move stuff around? (ahh...is this all static?)

Reply:
We also use rpath to make it more solid. Outputting also readelf so you can review.

```
readelf -d /gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/neurodamus-neocortex-develop-mc2tue/bin/special-core

Dynamic section at offset 0x47d20 contains 38 entries:
  Tag        Type                         Name/Value
 0x0000000000000001 (NEEDED)             Shared library: [libcorenrnmech_neocortex-develop.so]
 0x0000000000000001 (NEEDED)             Shared library: [libcoreneuron.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libreportinglib.so.2]
 0x0000000000000001 (NEEDED)             Shared library: [libhdf5.so.101]
 0x0000000000000001 (NEEDED)             Shared library: [libz.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libsyn2.so]
 0x0000000000000001 (NEEDED)             Shared library: [libpthread.so.0]
 0x0000000000000001 (NEEDED)             Shared library: [libmpi++abi1002.so]
 0x0000000000000001 (NEEDED)             Shared library: [/opt/hpe/hpc/mpt/mpt-2.16/lib/libmpi.so]
 0x0000000000000001 (NEEDED)             Shared library: [libstdc++.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libm.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libgcc_s.so.1]
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.6]
 0x0000000000000001 (NEEDED)             Shared library: [libdl.so.2]
 0x000000000000000f (RPATH)              Library rpath: [/gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/intel-18.0.3-kwbptw7qxx/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64:/gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/gcc-6.4.0-i6lyqfscua/lib:/gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/gcc-6.4.0-i6lyqfscua/lib64:$ORIGIN:/gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/coreneuron-develop-bmyahd/lib64:/gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/neurodamus-neocortex-develop-mc2tue/lib:/gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/08-08-2018/install/linux-rhel7-x86_64/intel-18.0.1/reportinglib-develop-uasv33/lib:/gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/hdf5-1.10.2-dueuws/lib:/gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/2018-09-28/install/linux-rhel7-x86_64/gcc-6.4.0/zlib-1.2.11-w43e56/lib:/gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/synapsetool-0.4.1-kpne43/lib64:/gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/neurodamus-neocortex-develop-mc2tue/lib64:/gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/boost-1.67.0-6skhmz/lib:/gpfs/bbp.cscs.ch/project/proj12/jenkins/devel_builds/install/linux-rhel7-x86_64/intel-18.0.1/neuron-2018-10-3a3zr2/lib:/opt/hpe/hpc/mpt/mpt-2.16/lib]
```


```
leite@bbpv2 ~/test/blueconfigs (master %)$ ldd /gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/neurodamus-neocortex-develop-mc2tue/bin/special-core
	linux-vdso.so.1 =>  (0x00007fffedb05000)
	libcorenrnmech_neocortex-develop.so => /gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/neurodamus-neocortex-develop-mc2tue/lib/libcorenrnmech_neocortex-develop.so (0x00007fffed8ad000)
	libcoreneuron.so.1 => /gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/coreneuron-develop-bmyahd/lib64/libcoreneuron.so.1 (0x00007fffed5b2000)
	libreportinglib.so.2 => /gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/08-08-2018/install/linux-rhel7-x86_64/intel-18.0.1/reportinglib-develop-uasv33/lib/libreportinglib.so.2 (0x00007fffed36c000)
	libhdf5.so.101 => /gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/hdf5-1.10.2-dueuws/lib/libhdf5.so.101 (0x00007fffecd7c000)
	libz.so.1 => /gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/2018-09-28/install/linux-rhel7-x86_64/gcc-6.4.0/zlib-1.2.11-w43e56/lib/libz.so.1 (0x00007fffecb65000)
	libsyn2.so => /gpfs/bbp.cscs.ch/home/leite/test/blueconfigs/INSTALL_HOME/install/linux-rhel7-x86_64/intel-18.0.3/synapsetool-0.4.1-kpne43/lib64/libsyn2.so (0x00007fffec87f000)
	libpthread.so.0 => /usr/lib64/libpthread.so.0 (0x00007fffec62f000)
	libmpi++abi1002.so => /opt/hpe/hpc/mpt/mpt-2.16/lib/libmpi++abi1002.so (0x00007fffec3f4000)
	/opt/hpe/hpc/mpt/mpt-2.16/lib/libmpi.so (0x00007fffec01f000)
	libstdc++.so.6 => /gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/gcc-6.4.0-i6lyqfscua/lib64/libstdc++.so.6 (0x00007fffebc9d000)
	libm.so.6 => /usr/lib64/libm.so.6 (0x00007fffeb99b000)
	libgcc_s.so.1 => /gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/gcc-6.4.0-i6lyqfscua/lib64/libgcc_s.so.1 (0x00007fffeb784000)
	libc.so.6 => /usr/lib64/libc.so.6 (0x00007fffeb3c0000)
	libdl.so.2 => /usr/lib64/libdl.so.2 (0x00007fffeb1bc000)
	libimf.so => /gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/intel-18.0.3-kwbptw7qxx/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libimf.so (0x00007fffeac28000)
	libsvml.so => /gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/intel-18.0.3-kwbptw7qxx/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libsvml.so (0x00007fffe931a000)
	libirng.so => /gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/intel-18.0.3-kwbptw7qxx/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libirng.so (0x00007fffe8fa5000)
	libintlc.so.5 => /gpfs/bbp.cscs.ch/apps/hpc/jenkins/deploy/compilers/2018-12-18/linux-rhel7-x86_64/gcc-4.8.5/intel-18.0.3-kwbptw7qxx/compilers_and_libraries_2018.3.222/linux/compiler/lib/intel64/libintlc.so.5 (0x00007fffe8d37000)
	libsz.so.2 => /gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/libszip-2.1.1-gszkr7/lib/libsz.so.2 (0x00007fffe8b21000)
	libcpuset.so.1 => /usr/lib64/libcpuset.so.1 (0x00007fffe8914000)
	libbitmask.so.1 => /usr/lib64/libbitmask.so.1 (0x00007fffe870f000)
	libboost_filesystem.so.1.67.0 => /gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/boost-1.67.0-6skhmz/lib/libboost_filesystem.so.1.67.0 (0x00007fffe84f3000)
	libboost_system.so.1.67.0 => /gpfs/bbp.cscs.ch/apps/hpc/spack-deployments/18-06-2018/install/linux-rhel7-x86_64/gcc-6.4.0/boost-1.67.0-6skhmz/lib/libboost_system.so.1.67.0 (0x00007fffe82ed000)
	/lib64/ld-linux-x86-64.so.2 (0x0000555555554000)
	librt.so.1 => /usr/lib64/librt.so.1 (0x00007fffe80e5000)
```